report-ng/app/src/components/timings/test-timings/test-timings.ts

```diff
@@ -0,0 +1,357 @@
+
+@autoinject()
+export class TestTimings extends AbstractViewModel {
+    private static readonly TEST_NUMBER_LIMIT = 20;
+    private static readonly TEST_COLOR = '#6897EA';
+    private static readonly TEST_COLOR_PALE = '#c8d4f4';
+    private _chart: ECharts;
+    private _executionStatistics: ExecutionStatistics;
+    private _option: EChartsOption;
+    private _hasEnded = false;
+    private _methodDetails: MethodDetails[];
+    private _labels: string[];
+    private _sectionValues: number[];
+    private _data: number[];
+    private _bars: IDurationBar[];
+    private _loading = false;
+    private _searchRegexp: RegExp;
+    private _inputValue = '';
+    private _methodId: string;
+    private _rangeOptions = ['5', '10', '15', '20']; // ranges of time in seconds that the test durations can be sorted in
+    private _showConfigurationMethods: boolean = null;
+    private _lookUpOptions;
+    @observable private selectedOptionId;
+    private _methodNameInput;
+
+    constructor(
+        private _statusConverter: StatusConverter,
+        private _statisticsGenerator: StatisticsGenerator,
+    ) {
+        super();
+        this._bars = [];
+        this._lookUpOptions = [];
+        this._option = {};
+    }
+
+    activate(params: any, routeConfig: RouteConfig, navInstruction: NavigationInstruction) {
+        super.activate(params, routeConfig, navInstruction);
+        if (!this.queryParams.rangeNum) this.queryParams.rangeNum = '10';// only set range 10 if there is no range at all (e.g. when navigation from another view)
+        if (params.config) {
+            this._showConfigurationMethods = !!params.config.toLowerCase();
+        }
+        this._getLookUpOptions()
+        this._filter();
+    }
+
+    selectedOptionIdChanged(methodId: string) {
+        if (methodId) {
+            this._methodId = methodId;
+            this.queryParams.methodId = methodId
+            this.updateUrl(this.queryParams);
+            this._highlightData()
+        }
+    }
+
+    selectionChanged() { // called when input from lookup text field is deleted manually
+        this._setChartOption(); // overwrites color highlighting (reset)
+        if (this._inputValue.length == 0) {
+            this._methodId = undefined;
+            delete this.queryParams.methodId;
+            this.updateUrl({rangeNum: this.queryParams.rangeNum, methodId: this._methodId});
+            this._getLookUpOptions();
+        }
+    }
+
+    private _getLookUpOptions() {
+        this._statisticsGenerator.getExecutionStatistics().then(executionStatistics => {
+            let methodContexts: IMethodContext[];
+            if (this.queryParams.methodId) {
+                methodContexts = [executionStatistics.executionAggregate.methodContexts[this.queryParams.methodId]];
+                this._highlightData();
+                this.updateUrl({methodId: this.queryParams.methodId});
+                this._methodId = this.queryParams.methodId;
```

Comment:
Please consider getting rid of `this._methodId` like you did for `this._rangeNum`